packages/scm-extra/src/browser/history/scm-history-widget.tsx

```diff
@@ -189,19 +188,17 @@ export class ScmHistoryWidget extends ScmNavigableListWidget<ScmHistoryListNode>
         if (repository) {
             if (this.historySupport) {
                 try {
-                    const currentCommits = this.status.state === 'ready' ? this.status.commits : [];
-
-                    let history = await this.historySupport.getCommitHistory(options);
+                    const history = await this.historySupport.getCommitHistory(options);
                     if (token.isCancellationRequested || !this.hasMoreCommits) {
                         return;
                     }
 
-                    if (options && ((options.maxCount && history.length < options.maxCount) || (!options.maxCount && currentCommits))) {
+                    if (options && (options.maxCount && history.length < options.maxCount)) {
                         this.hasMoreCommits = false;
                     }
-                    if (currentCommits.length > 0) {
-                        history = history.slice(1);
-                    }
+
+                    const avatarCache = new Map<string, string>();
```

Comment:
Your code never calls `set` on this cache. Is it an oversight, or am I missing something?